src/app/items/effects/items.effects.ts

```diff
@@ -25,44 +27,50 @@ export class ItemsEffects {
     ofType(initiateItems),
     switchMap(() => this.itemsService.getAll$()
       .pipe(
-        map((items: Item[]) => initiateItemsSuccess({items})),
-        catchError((err) => of(initiateItemsFail({message: err})))
+        map((items: Item[]) => initiateItemsSuccess({ items })),
+        catchError((err) => of(initiateItemsFail({ message: err })))
       ))
-    )
+  )
   );
 
   deleteItem$ = createEffect(() => this.actions$.pipe(
     ofType(deleteItem),
-    switchMap(({itemToDelete}) => this.itemsService.delete$(itemToDelete)
+    switchMap(({ itemToDelete }) => this.itemsService.delete$(itemToDelete)
       .pipe(
-        map(() => deleteItemSuccess({deletedItemId: itemToDelete.id})),
-        catchError((err) => of(deleteItemFail({message: err})))
+        map(() => deleteItemSuccess({ deletedItemId: itemToDelete.id })),
+        catchError((err) => of(deleteItemFail({ message: err })))
       ))
-    )
+  )
   );
 
   updateItem$ = createEffect(() => this.actions$.pipe(
     ofType(updateItem),
-    switchMap(({item}) => this.itemsService.upsert$(item)
-      .pipe(
-        map(() => updateItemSuccess({updatedItem: item})),
-        catchError((err) => of(updateItemFail({message: err})))
-      ))
-    )
+    withLatestFrom(this.store$.select(getUser)),
```

Comment:
it is better to use `this.store.pipe(select(getUser))`
just newer syntax while the old one will be deprecated